tests/core/data/test_json_data_node.py

```diff
@@ -159,35 +171,24 @@ def test_write_non_serializable(self, json_file):
     def test_write_date(self, json_file):
         json_dn = JSONDataNode("foo", Scope.PIPELINE, properties={"default_path": json_file})
         now = datetime.datetime.now()
-        now_str = now.isoformat()
         data = {"date": now}
         json_dn.write(data)
         read_data = json_dn.read()
-        assert read_data["date"] == now_str
+        assert read_data["date"] == now
 
     def test_write_enum(self, json_file):
-        class MyEnum(Enum):
-            A = 1
-            B = 2
-            C = 3
-
         json_dn = JSONDataNode("foo", Scope.PIPELINE, properties={"default_path": json_file})
         data = [MyEnum.A, MyEnum.B, MyEnum.C]
         json_dn.write(data)
         read_data = json_dn.read()
-        assert read_data == [1, 2, 3]
+        assert read_data == [MyEnum.A, MyEnum.B, MyEnum.C]
 
```

Comment:
Great! Do you think we can also add the dictionary to make sure we cover as much data type in the test as possible?

Reply:
The `test_write()` test the write method with a dictionary, so I think we're good on this